src/Engines/BaseEngine.cs

```diff
@@ -236,43 +236,57 @@ internal MimeBundle EncodeForDisplay(object displayable)
 
         private void WriteToStream(Message parent, StreamName stream, string text)
         {
-            // Send the engine's output to stdout.
-            this.ShellServer.SendIoPubMessage(
-                new Message
-                {
-                    Header = new MessageHeader
-                    {
-                        MessageType = "stream"
-                    },
-                    Content = new StreamContent
+            try
+            {
+                // Send the engine's output to stdout.
+                this.ShellServer.SendIoPubMessage(
+                    new Message
                     {
-                        StreamName = stream,
-                        Text = text
-                    }
-                }.AsReplyTo(parent)
-            );
+                        Header = new MessageHeader
+                        {
+                            MessageType = "stream"
+                        },
+                        Content = new StreamContent
+                        {
+                            StreamName = stream,
+                            Text = text
+                        }
+                    }.AsReplyTo(parent)
+                );
+            }
+            catch(Exception e)
+            {
+                this.Logger?.LogError(e, "Unexpected error when trying to write to stream.");
+            }
         }
 
         private void WriteDisplayData(Message parent, object displayable)
         {
-            if (displayable == null) throw new ArgumentNullException(nameof(displayable));
-            var serialized = EncodeForDisplay(displayable);
-            // Send the engine's output to stdout.
-            this.ShellServer.SendIoPubMessage(
-                new Message
-                {
-                    Header = new MessageHeader
-                    {
-                        MessageType = "display_data"
-                    },
-                    Content = new DisplayDataContent
+            try
+            {
+                if (displayable == null) throw new ArgumentNullException(nameof(displayable));
```

Comment:
Should this line be outside the `try`/`catch` so that we can find `null` inputs?

Reply:
Thanks for bringing this up.
I looked into the places where the code was throwing for null arguments and I believe we were being to restrictive on not allowing null arguments. In particular, Encoders should support encoding null data (simply ignore it), after all, `null` is a valid result from calling an operation.
So I changed the code in the `BaseEngine` to be a lot more liberal for `null` arguments, which makes it a lot more robust -it was crashing very easily-. 
Could you please take a new look?